Recrutify/Recrutify.Services/Services/ScheduleService.cs

```diff
@@ -80,9 +92,80 @@ public async Task UpdateScheduleSlotsForCurrentUserAsync(IEnumerable<DateTime> d
             await _scheduleRepository.BulkUpdateScheduleSlotsAsync(currentUserId, newListDateTime, removedListDateTime);
         }
 
-        private IEnumerable<DateTime> GetDateTimeInScheduleSlots(IEnumerable<ScheduleSlot> scheduleSlots)
+        public async Task ProcessingAppointOrCancelInterviewsAsync(IEnumerable<InterviewDTO> interviews, Guid projectId, CancellationToken cancellationToken)
         {
-            return scheduleSlots?.Select(s => s.AvailableTime) ?? new List<DateTime>();
+            var projectExists = await _projectService.ExistsAsync(projectId, cancellationToken);
+            if (!projectExists)
+            {
+                throw new ValidationException("Project does not exist.");
+            }
+
+            var allCandidates = await _candidateService.GetCandidatesByIdsAsync(interviews.Select(i => i.CandidateId));
+            var dictionaryInterviews = _mapper.Map<IEnumerable<Interview>>(interviews)
+                                                .GroupBy(i => i.IsAppointment)
+                                                .ToDictionary(k => k.Key, v => v.Select(i => i).ToList());
+
+            var canceledInterviews = dictionaryInterviews.TryGetValue(false, out var i) ? i : default;
+            var appointedtInterviews = dictionaryInterviews.TryGetValue(true, out var l) ? l : default;
+
+            if (canceledInterviews != null)
+            {
+                await BulkСancelInterviewsAsync(projectId, allCandidates.Where(c => canceledInterviews.Select(i => i.CandidateId).Contains(c.Id)), canceledInterviews);
+            }
+
+            if (appointedtInterviews != null)
+            {
+                await BulkAppointInterviewsAsync(projectId, allCandidates.Where(c => appointedtInterviews.Select(i => i.CandidateId).Contains(c.Id)), appointedtInterviews);
+            }
         }
+
+        private async Task BulkСancelInterviewsAsync(Guid projectId, IEnumerable<Candidate> candidates, IEnumerable<Interview> interviews)
+        {
+            var cuurentStatusCadidates = candidates.FirstOrDefault().ProjectResults.FirstOrDefault(p => p.ProjectId == projectId).Status;
+
+            if (!interviews.Any())
```

Comment:
Move to the beginning of method, there is a case when you do prev line without sense

Reply:
+